admin-writer/src/main/resources/raw_data/db/migration/R__2_Mat_Views.sql

```diff
@@ -185,4 +185,238 @@ SELECT
+JOIN raw_data.view_builds builds ON builds.build_id = coverage.build_id
+ORDER BY coverage.session_started_at;
+
+-----------------------------------------------------------------
+
+-----------------------------------------------------------------
+DROP MATERIALIZED VIEW IF EXISTS raw_data.matview_test_session_build_coverage CASCADE;
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS raw_data.matview_test_session_build_coverage AS
+SELECT
+	s.test_session_id,
+	s.group_id,
+	s.test_task_id,
+	s.started_at,
+	s.tests,
+	s.result,
+	s.duration,
+	s.failed,
+	s.passed,
+	s.skipped,
+	s.smart_skipped,
+	s.success,
+	s.successful,
+	sb.build_id,
+	sb.build_tests,
+	sc.covered_probes,
+	sc.unique_covered_probes,
+	sc.accumulated_covered_probes,
+	sc.probes_coverage_ratio,
+	sc.unique_probes_coverage_ratio,
+	sc.accumulated_probes_coverage_ratio,
+	sc.isolated_covered_probes,
+	sc.isolated_unique_covered_probes,
+	sc.isolated_accumulated_covered_probes,
+	sc.isolated_probes_coverage_ratio,
+	sc.isolated_unique_probes_coverage_ratio,
+	sc.isolated_accumulated_probes_coverage_ratio
+FROM raw_data.view_test_sessions s
+LEFT JOIN raw_data.view_test_session_builds_v2 sb ON sb.test_session_id = s.test_session_id
+LEFT JOIN raw_data.view_test_session_coverage sc ON sc.test_session_id = s.test_session_id AND sc.build_id = sb.build_id;
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_matview_test_session_build_coverage_pk ON raw_data.matview_test_session_build_coverage (test_session_id, build_id);
+
+-----------------------------------------------------------------
+
+-----------------------------------------------------------------
+DROP MATERIALIZED VIEW IF EXISTS raw_data.matview_builds_comparison CASCADE;
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS raw_data.matview_builds_comparison AS
+WITH
+    BaselineMethods AS (
+        SELECT
+            m.group_id,
+            m.app_id,
+            m.build_id,
+            m.signature,
+            m.body_checksum,
+            m.probes_count
+        FROM raw_data.matview_methods_with_rules m
+    ),
+    TargetMethods AS (
+        SELECT
+            m.group_id,
+            m.app_id,
+            m.build_id,
+            m.signature,
+            m.body_checksum,
+            m.probes_count,
+            (SUM(m.probes_count) OVER (PARTITION BY m.build_id ORDER BY m.signature)) - m.probes_count + 1 AS probes_start,
+            (COUNT(*) OVER (PARTITION BY m.build_id ORDER BY m.signature)) AS method_num
+        FROM raw_data.matview_methods_with_rules m
+    ),
+    MethodsComparison AS (
+        SELECT
+            m.build_id,
+            m.signature,
+            m.body_checksum,
+            m.probes_count,
+            m.probes_start,
+            m.method_num,
+            baseline.build_id AS baseline_build_id,
+            baseline.body_checksum AS baseline_body_checksum,
+            (CASE WHEN m.body_checksum = baseline.body_checksum AND m.probes_count = baseline.probes_count THEN 1 ELSE 0 END) AS equal,
```

Comment:
whats the motivation for checking probes count? Isn't checksum equality check sufficient?

Reply:
Yes, it is important to take into account both the method checksum and the probes count, because we collect all the probes of equivalent methods into one long build probe. If the methods have the same checksum, but the number of probes does not match, then the probes of the same build will be of different lengths. This will break the calculation of aggregated coverage for builds.